modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -119,6 +119,9 @@ protected ReindexRequest buildRequest(RestRequest request) throws IOException {
         try (XContentParser parser = request.contentParser()) {
             PARSER.parse(parser, internal, null);
         }
+        if (request.hasParam("scroll")) {
```

Comment:
Since you changed `AbstractBulkByScrollRequest` to accept a scroll parameter you could parse this option directly in `AbstractBaseReindexRestHandler#setCommonOptions`. This way all `reindex` actions would be able to set this option (`_delete_by_query`, _update_by_query`).

Reply:
Thx for your quick review. In fact I see that for `_delete_by_query` and `_update_by_query`, we have already parsed the rest request here in their superclass https://github.com/elastic/elasticsearch/blob/master/modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractBulkByQueryRestHandler.java#L54 by calling `RestSearchAction#parseSearchRequest`, so also including `scroll` parameter. I'm not sure for `RestReindexAction` we should do the same thing or not (as `_delete_by_query` and `_update_by_query` have a superclass of their own) ? So I just added it in `RestReindexAction`.

Reply:
You're right it's not needed for `_delete_by_query` and `_update_by_query` since they parse the request with `RestSearchAction`. I missed this part so I think that this pr is ready and fixes the main issue; adding the support to change the default `scroll` for `_reindex` request.